test/bfe-nil-msg.test.js

```diff
@@ -0,0 +1,24 @@
+const tape = require('tape')
+const bfe = require('ssb-bfe')
+const convertNilForEnvelopeSpec = require('../util/bfe-nil-msg')
+
+tape('nil envelope', function (t) {
+  const bbFeed = '@6CAxOI3f+LUOVrbAl0IemqiS7ATpQvr9Mdw9LC4+Uv0=.bbfeed-v1'
+  const bbNull = bfe.convertNilForEnvelopeSpec(bfe.encode(null), bfe.encode(bbFeed))
+  t.equal(bbNull.slice(0,2).toString('hex'), '0104', 'bendy msg')
+  t.ok(bbNull.slice(2).equals(Buffer.alloc(32)), 'ends with 32 zeros')
+  t.equal(bbNull.length, 32 + 2, 'length ok')
+
+  const classicFeed = '@6CAxOI3f+LUOVrbAl0IemqiS7ATpQvr9Mdw9LC4+Uv0=.ed25519'
+  const classicNull = bfe.convertNilForEnvelopeSpec(bfe.encode(null), bfe.encode(classicFeed))
+  t.equal(classicNull.slice(0,2).toString('hex'), '0100', 'classic msg')
+  t.ok(classicNull.slice(2).equals(Buffer.alloc(32)), 'ends with 32 zeros')
+  t.equal(classicNull.length, 32 + 2, 'length ok')
+
+  const msg = '%HZVnEzm0NgoSVfG0Hx4gMFbMMHhFvhJsG2zK/pijYII=.sha256'
+  const classicMsg = bfe.convertNilForEnvelopeSpec(bfe.encode(msg), bfe.encode(classicFeed))
+  t.equal(classicMsg.slice(0,2).toString('hex'), '0100', 'classic msg')
+  t.notOk(classicMsg.slice(2).equals(Buffer.alloc(32)), 'is not null')
+  t.equal(bfe.decode(classicMsg), msg, 'decodes back to the same')
+  t.end()
```

Comment:
I don't understand why this test is in this repo?
Also why is it that you would ever want an empty data part for these types?

 # :fire: (i.e. danger)

I feel like you should not be allowed to make zero-data / zero-key feed/msg?
Seems like a good way to shoot yourself in the foot.
BUUUT, I also don't have the full context loaded into my head

Reply:
This is the null case I alluded to. Before (in ssb-bfe) when encoding a message, say `{ subfeed: x.ed25519, metafeed: y.bbfeed-v1, previous: null }` you have to use the correct encoder as it's not really possible to detect from the content what previous is here. This is error prone and not super general. Instead now you just call encode and decode and it works. We didn't add this kind of conversion code to bfe as it is not really a concern of that library. Bendy butt works perfectly fine with the same nil message encoding for classic and bendy butt messages. I came from reading the TFK spec first so I thought this wouldn't work, but we really only need to handle it here, in ssb-tribes there are no changes related to this.

Reply:
Ah yeah, I remember. This is about the specific very unlikely edge case that prev_msg_id is null
(unlikely because starting with an encrypted message is fairly rare, but not impossible)
Yes?

Reply:
I don't understand the Meta feed context enough to fully grok what you've said.

Reply:
yes